### PR TITLE
added ActiveRecord like logging messages for each executed migration

### PR DESCRIPTION
**Pull request by alubbe**
see https://github.com/sequelize/cli/issues/92

### CODE REVIEW

**Comment by rpaterson** [comment]
Please don't output directly to console.log - use some logging system so that clients can control where the log goes.  E.g. Sequelize accepts an option { logging: <function> } and uses that to output logs.

**Comment by alubbe** [comment]
does umzug already have a ```logging:``` option?
afaik, the sequelize-cli disables the sequelize logs (mainly to disable the sql prints), so I don't think I can use their logging function.

What would you propose?

**Comment by rpaterson** [comment]
I'm not suggesting using the sequelize logs directly.  What I meant was add a `logging:` option to Umzug if it doesn't have one already.  Sequelize/cli can specify `logging: console.log` and other clients of Umzug can do as they see fit.

**Comment by alubbe** [comment]
I like that idea - see newest commit

File: index.js [hunk 1]
```diff
@@ -184,6 +207,12 @@ var Umzug = module.exports = redefine.Class({
     }
   },
 
+  log: function() {
+    if (this.options.logging) {
+      this.options.logging.apply(null, [].slice.call(arguments));
```
Comment (rpaterson) [inline index.js]: What do you expect to happen when the logging method is called with multiple arguments?

**Comment by rpaterson** [comment]
Looks good to me, just missing documentation.

**Comment by sdepold** [comment]
Will review it later today

File: index.js (hunk 1 above)
Comment (alubbe) [inline index.js]: I'm not sure I fully understand your question. The consumer could potentially add their very own logging function that takes multiple arguments, or even want to pass multiple arguments into console.log. In both cases I would expect that .log passes them on 1-1 to the passed logging function. Does that answer your question?

**Comment by alubbe** [comment]
@rpaterson does the documentation just go into the readme or somewhere else, as well? Do you want me to write something up?

File: index.js (hunk 1 above)
Comment (rpaterson) [inline index.js]: A user of Umzug who is writing a logging method needs to know what to do with multiple arguments - e.g. concat them all with .toString?  Or is the first argument a context object, the second a template, and subsequent args params as per Bunyan logging?  Conversely the Umzug code needs to know when and how to call the log method with multiple args.  I think probably easiest would be just to have one argument that is a string - that's all you're using today AFAICT anyway.

**Comment by rpaterson** [comment]
Just the readme AFAIK

File: index.js (hunk 1 above)
Comment (alubbe) [inline index.js]: understood and agreed. let's simplify for now and add more arguments, such as a possible options arguments, only when actually needed

**Comment by sdepold** [comment]
Looks good. The readme is right. Please also add tests for the log function :)

**Comment by alubbe** [comment]
doc & tests added

**Comment by rpaterson** [comment]
LGTM

**Comment by sdepold** [comment]
Sorry to bother you but would it be possible to actually call the migrate function and check if the expected logging behavior is executed?

**Comment by alubbe** [comment]
Okay here we go

**Comment by alubbe** [comment]
@sdepold feedback?

**Comment by sdepold** [comment]
Sorry. The last days were a bit stressful :) Could you please add an assertion for the actual arguments that get passed to logger? So ideally we would really make messages are getting logged.

**Comment by sdepold** [comment]
Hossa! Merged + added the missing test pieces on my own.

**Comment by sdepold** [comment]
just released v1.5.0

**Comment by alubbe** [comment]
awesome!

**Pull request merged**